assets/app/scripts/directives/nav.js

```diff
@@ -53,11 +53,16 @@ angular.module('openshiftConsole')
 
           sortedProjects = $filter('orderByDisplayName')(projects);
 
-          options = _.map(sortedProjects, function(item) {
+          options = _.map(sortedProjects, function(item, index) {
```

Comment:
That's a lot of calls to the display name filter. Probably faster (and easier to follow) to build a hash of displayNames:true, then just check against that when building the options. Also, don't show in parens if the display name and project name are the same